var/spack/repos/builtin/packages/sgpp/package.py

```diff
@@ -52,10 +54,22 @@ class Sgpp(SConsPackage):
     # to complain about missing headers (due to a path check not working anymore)
     # See issue https://github.com/SGpp/SGpp/issues/263 and https://github.com/SGpp/SGpp/pull/266
     patch("disable_disutils_deprecation_warning.patch", when="@:3.4.0 ^python@3.10:3.11")
-    # SGpp does not contain aarch64 support as of yet. To make it work still, this patch adds
+    # SGpp does not contain aarch64 support as of 3.4.0. To make it work still, this patch adds
     # simple build system support for it.
-    patch("for_aarch64.patch", when="target=aarch64:")
+    patch("for_aarch64.patch", when="@:3.4.0 target=aarch64:")
+    # SGpp will default to the system paths when linking boost without the patch
+    # This may work (depending on the boost versions in question) but we should use the boost
+    # from spack. This patch allows to correctly pass the spack's boost path to SGpp
+    # Fixed in SGpp PR https://github.com/SGpp/SGpp/pull/273
+    patch("set_boost_lib_path_internally.patch", when="@3.3.0:3.4.0")
```

Comment:
Same comment as above. Only applies if you expect patch versions:
```suggestion
    patch("set_boost_lib_path_internally.patch", when="@3.3:3.4")
```

Edit: same applies to the other 3.4.0 version ranges. Feel free to ignore if it doesn't apply.